### ISSUE TITLE
Which Kubernetes Permissions does a user need?

### ISSUE THREAD

**Issue by torbendury** [fission/fission#2443]
Hello,

since we're relying on Kubernetes RBAC, it is atm my chosen path to create Kubernetes ServiceAccounts (like a technical user) to let developers deploy their functions via a deployment pipeline.

To let them do so, I need to know which RBAC permissions are the bare minimum to deploy a full fledged fission function (including routes, triggers, ...).

Is there any documentation for it? I couldn't manage to find one.

Thanks for your help in advance!

**Comment by torbendury** [comment]
Hi @sanketsudake and thanks for your answer! I see that this is a cluster-wide role that would lead to cluster-wide permissions. Is it possible, for a developer `ServiceAccount`, to cut those permissions down to a namespace? Would they need to be present in `fission`, `fission-function` and `fission-builder` namespaces?

Also, the permissions seem quite elevated for **only** creating functions and the respective triggers/routes/ingresses. Are we able to cut them down to the bare minimum?

**Comment by sanketsudake** [comment]
We have trimmed down lot of permissions per Fission component. 

- We have separate service accounts per fission component.
- We are converting most of the cluster roles to roles and creating them per namespace user configured.
- We have also removed the need of fission-function and fission-builder 

There are still few changes pending in the series, but we will complete them by the time the 1.18 release is available.

https://fission.io/docs/releases/v1.18.0-rc1/

CLI permissions required
https://fission.io/docs/usage/rbac-permissions/
